src/mqt/predictor/ml/predictor.py

```diff
@@ -356,171 +343,89 @@ def get_prepared_training_data(
 
         Returns:
-            The probabilities for all supported quantum devices to be the most suitable one for the given quantum circuit.
+           The training data, the names list and the scores list.
         """
-        if self.clf is None:
-            path = ml.helper.get_path_trained_model(figure_of_merit)
-            if path.is_file():
-                self.clf = load(path)
-
-            if self.clf is None:
-                error_msg = "The ML model is not trained yet. Please train the model before using it."
-                logger.error(error_msg)
+        with resources.as_file(ml.helper.get_path_training_data() / "training_data_aggregated") as path:
+            if (
+                path.joinpath("training_data_" + self.figure_of_merit + ".npy").is_file()
+                and path.joinpath("names_list_" + self.figure_of_merit + ".npy").is_file()
+                and path.joinpath("scores_list_" + self.figure_of_merit + ".npy").is_file()
+            ):
+                training_data = np.load(path / ("training_data_" + self.figure_of_merit + ".npy"), allow_pickle=True)
+                names_list = list(np.load(path / ("names_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
+                scores_list = list(np.load(path / ("scores_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
+            else:
+                error_msg = "Training data not found. Please run the training script first."
                 raise FileNotFoundError(error_msg)
 
-        feature_dict = ml.helper.create_feature_dict(qc)  # type: ignore[unreachable]
-        feature_vector = list(feature_dict.values())
-
-        path = ml.helper.get_path_trained_model(figure_of_merit, return_non_zero_indices=True)
-        non_zero_indices = np.load(path, allow_pickle=True)
-        feature_vector = [feature_vector[i] for i in non_zero_indices]
-
-        return self.clf.predict_proba([feature_vector])[0]
+            return training_data, names_list, scores_list
+
+
+def predict_device_for_figure_of_merit(
+    qc: Path | QuantumCircuit, figure_of_merit: reward.figure_of_merit = "expected_fidelity"
+) -> Device:
+    """Returns the probabilities for all supported quantum devices to be the most suitable one for the given quantum circuit.
+
+    Arguments:
+        qc: The QuantumCircuit or Path to the respective qasm file.
+        figure_of_merit: The figure of merit to be used for compilation.
+
+    Returns:
+        The probabilities for all supported quantum devices to be the most suitable one for the given quantum circuit.
+    """
+    path = ml.helper.get_path_trained_model(figure_of_merit)
+    if not path.exists():
+        error_msg = "The ML model is not trained yet. Please train the model before using it."
+        logger.error(error_msg)
+        raise FileNotFoundError(error_msg)
+    clf = load(path)
+    logger.warning(load(path))
+    logger.warning(path)
```

Comment:
Why are you logging warnings here? At best, these should be debug info prints.

Reply:
That was a left-over from debugging, removed it now.